src/ai/backend/manager/idle.py

```diff
@@ -655,7 +669,13 @@ async def get_extra_info(self, session_id: SessionId) -> Optional[dict[str, Any]
         return None
 
     async def check_idleness(
-        self, kernel: Row, dbconn: SAConnection, policy: Row, redis_obj: RedisConnectionInfo
+        self,
+        kernel: Row,
+        dbconn: SAConnection,
+        policy: Row,
+        redis_obj: RedisConnectionInfo,
+        *,
+        grace_period_end: Optional[datetime],
```

Comment:
Did it meant to be a keyword parameter? If then please add default value. 
```suggestion
        grace_period_end: Optional[datetime] = None,
```